### ISSUE TITLE
Feature Request: Add early stopping on test set for `surv.xgboost`

### ISSUE THREAD

**Issue by bblodfon** [mlr-org/mlr3extralearners#250]
Hi!

I saw that recently support for having a test set for early stopping was added in `classif.xgboost` and `regr.xgboost` learners, see [here](https://github.com/mlr-org/mlr3learners/pull/234/). It would be amazing if we had this also for `surv_xgboost`! I will give it a try to implement, should be doable :)

**Comment by sebffischer** [comment]
Yes good idea! :)

**Comment by sebffischer** [comment]
lmk if you need help

**Comment by bblodfon** [comment]
I have some questions:

- In the [test](https://github.com/bblodfon/mlr3extralearners/blob/main/tests/testthat/test_xgboost_surv_xgboost.R) I wrote, is it okay to use `tsk('lung')` from `mlr3proba`? where is that loaded from? I included `mlr3pipelines::po()` is that okay?
- What does happen when you run in the test `learner$train(task, rows = 1:200)` (which works all fine)? Internally, does it use for training more than the 80% - 182 rows that have been assigned the `use` role and so some of these overlap with the ones assigned the `test` role (but these should be mutually exclusive, right)?

**Comment by bblodfon** [comment]
@sebffischer sorry for bothering you again about this, isn't point 2 above pointing to unexpected behavior? (point 1 was about adding the package to `Suggests`). It refers to this test:
https://github.com/bblodfon/mlr3extralearners/blob/main/tests/testthat/test_xgboost_surv_xgboost.R#L34

**Comment by sebffischer** [comment]
Sorry, I forgot that this thread here existed. 

I wouldn't call this bothering at all, it is a very valid question :) 
In generally, we don't enforce that the `test` and `train` rows are mutually exclusive. There is for examle also the `ResamplingInsample`, which uses the same data for training and testing. 
What would happen in the case you suggested is that the first 200 rows are used for training and `split$test` is used for testing.

**Comment by sebffischer** [comment]
So to your answer: Yes they should be mutually exclusive but we don't stop you from doing something else if you insist.

**Comment by bblodfon** [comment]
Thanks for making it clear!

**Comment by bblodfon** [comment]
Implemented in PR https://github.com/mlr-org/mlr3extralearners/pull/251
